lib/src/platform/src/realtime/connection.dart

```diff
@@ -24,6 +24,7 @@ class Connection extends PlatformObject {
     on().listen((event) {
       _state = event.current;
       _errorReason = event.reason;
+      invoke<String?>(PlatformMethod.connectionId).then((_id) => id = _id);
```

Comment:
Could you move this back? This way, id will be updated with a delay, which isn't good. I suggested that we send a syntactic event on the platform side with the id (not only the state), subscribe to it here on the syntactic event, and retrieve the id from the event.

Reply:
This is not a better solution either. We still can't guarantee delay in the event. Also, `ConnectionEvent` doesn't include `id` according to the spec.

Reply:
I feel it's better if we keep `id` as async

Reply:
https://sdk.ably.com/builds/ably/specification/main/features/#RTN4:~:text=enum%20ConnectionEvent

Reply:
Flutter SDK is a hack itself :) I believe it's very transparent solution, we can guarantee, that when connection state is become connected, `id` field will be set (what else do you think sdk user may expect? 

>ConnectionEvent doesn't include id according to the spec => https://sdk.ably.com/builds/ably/specification/main/features/#RTN4:~:text=enum%20ConnectionEvent

It doesn't matter it won't be spec connectionevent, it will be internal synthetic event. Public of course should be implemented according to the spec without id.

Reply:
updated